### PR TITLE
Reducer api context module

### PR DESCRIPTION
**Pull request by SJTGSHIVAM**
Added:
- reducer
- context
- interface
- API
- context-module
- some dummy UI files
- toastify

### CODE REVIEW

**Review by samyakshah3008 — APPROVED** [review]
Clean code, good folder structure and UI 
Good for approval ✅

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
 } from 'react-router-dom';
+import { ToastContainer } from 'react-toastify';
 
 export default function App() {
   return (
-    <Routes>
-      <Route path="/q" element={<Mockman />} />
-    </Routes>
+    <>
+      <header className="">
```
Comment (tejas-muthya37) [inline src/App.tsx:20]: You can remove the className attribute wherever it's not required.

**Review by tejas-muthya37 — APPROVED** [review]
Didn't find a lot of issues with the code, liked your way of destructuring the code, axios and BASE_API_URL.

**Review by Hnh013 — APPROVED** [review]
clean code and the folder structure is good

File: src/App.tsx (hunk 1 above)
Comment (SJTGSHIVAM) [reply src/App.tsx:20]: Hey, @tejas-muthya37  Thanks for pointing it out. I mentioned this is just the dummy code I left them as a placeholder. will fill them up with some meaningful full classes in the upcoming changes.

**Pull request merged**